packages/server/customer-os-api/repository/dashboard_repository.go

```diff
@@ -1528,3 +1529,121 @@ func getCorrectValueType(valueToExtract any) float64 {
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 9}) - duration({nanoseconds: 1}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 10}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 8}) - duration({nanoseconds: 1}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 9}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 7}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 8}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 6}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 7}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 5}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 6}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 4}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 5}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 3}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 4}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 2}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 3}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth - duration({months: 1}) AND sliItem.sliEndedAt > beginOfMonth - duration({months: 2}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END,
+						CASE WHEN sliItem.sliStartedAt < beginOfMonth AND sliItem.sliEndedAt > beginOfMonth + duration({months: 1}) - duration({nanoseconds: 1}) THEN sliItem.amount ELSE 0 END
+     ]) as sliActivePerContract,
+     REDUCE(s = [], sliItem in sliPerContract | 
+         CASE WHEN sliItem.sliStartedAt < startOfNextMonth  AND (sliItem.endedAt IS NULL or sliItem.endedAt > startOfNextMonth) THEN s + sliItem ELSE s END
+     ) as sliActiveInCurrentMonth
+WITH year, month, beginOfMonth, endOfMonth, cid, sliActivePerContract, sliActiveInCurrentMonth, 
+REDUCE(acc = null, i IN RANGE(0, 11) |
+    CASE WHEN sliActivePerContract[i] > 0 AND acc IS NULL THEN sliActivePerContract[i] ELSE acc END
+) AS baseline, 
+REDUCE(acc = 0, i IN sliActiveInCurrentMonth | acc + i.amount ) AS currentAmount
+WITH year, month, cid, baseline, currentAmount, sliActiveInCurrentMonth
+with year, month, SUM(baseline) as baselineValue, SUM(currentAmount) as currentValue
+return year, month, CASE WHEN baselineValue <> 0 THEN (currentValue / baselineValue) * 100 ELSE null END AS grossRevenueRetentionRate`, "% 12 + 1", tenant, tenant),
```

Comment:
@edifirut  I ve tested a lot and I found where the pain point is, but havent yet find a way to overcome it. 
`sliActiveInCurrentMonth` and `currentValue` seem working fine from the debug I did. 
The part that is not working as expected is the `baseline` on line 1612.
If you have any ideas on how to do it differently? 

`WITH year, month, beginOfMonth, endOfMonth, cid, sliActivePerContract, sliActiveInCurrentMonth, 
REDUCE(acc = null, i IN RANGE(0, 11) |
    CASE WHEN sliActivePerContract[i] > 0 AND acc IS NULL THEN sliActivePerContract[i] ELSE acc END
) AS baseline, `

Reply:
my suggestion would be to grab just this block and rework it in neo4j console. just build an array with arrays with values

WITH [
    [0,0,0,5,0,0,0,0,0,0,0,0],
    [0,2,0,0,0,0,0,0,0,0,0,0]
] as sliActivePerContract

MATCH (t:Tenant{name:"openlineai"})

return REDUCE(acc = null, i IN RANGE(0, 11) | CASE WHEN sliActivePerContract[i] > 0 AND acc IS NULL THEN sliActivePerContract[i] ELSE acc END ) AS baseline

Reply:
cool I ve worked it on the neo4j console. 

that is a working query for the above block 

`WITH [
  [0, 0, 0, 5, 0, 0, 0, 0, 0, 1, 0, 0],
  [0, 0, 0, 0, 0, 0, 29, 0, 0, 0, 0, 0],
  [1, 0, 0, 0, 0, 0, 29, 0, 0, 0, 0, 0]
] AS sliActivePerContract

WITH REDUCE(acc = 0, i IN RANGE(0, SIZE(sliActivePerContract) - 1) |
  acc + COALESCE(HEAD([value IN sliActivePerContract[i] WHERE value <> 0 | value]), 0)
) AS baseline

RETURN baseline;`

will try to add that to the repository now